kernel/comps/systree/src/utils.rs

```diff
@@ -31,8 +32,8 @@ impl SysObjFields {
         &self.id
     }
 
-    pub fn name(&self) -> &str {
-        self.name.deref()
+    pub fn name(&self) -> SysStr {
+        self.name.clone()
```

Comment:
We should not force the user to `clone` if it can be avoided. If a method does the `clone` internally, the user won't be able to avoid it.

```rust
    pub fn name(&self) -> &SysStr {
        &self.name
```

On the other hand, if the interface is defined as above, then the caller will have a choice on whether or not she wants to have a owned version.

Reply:
I would suggest to just return `&str` via [`Cow::as_ref`](https://doc.rust-lang.org/std/borrow/enum.Cow.html#method.as_ref).

A `&Cow` is a pointer to another pointer, which (i) does not make much sense and (ii) does not help the caller anything.

For example, if the caller wants a `String`, it will still need to clone. `&Cow` cannot help at all.

Reply:
> I would suggest to just return `&str` via [`Cow::as_ref`](https://doc.rust-lang.org/std/borrow/enum.Cow.html#method.as_ref).

Returning `&str` also works for me. This is the original implementation.

> (ii) does not help the caller anything.... For example, if the caller wants a String, it will still need to clone. &Cow cannot help at all.

It could be helpful if `SysStr` happens to be a `Cow::Borrowed(&'static str)`. In this case, the caller does not need to do a heap allocation in order to get an owned string.

> A &Cow is a pointer to another pointer, which (i) does not make much sense 

I think it is ok. From the caller's perspective, all she needs to deal with is a `&SysStr`, not a reference to a reference.